lib/mayaUsd/fileio/translators/translatorMayaReference.cpp

```diff
@@ -424,29 +472,74 @@ MStatus UsdMayaTranslatorMayaReference::update(const UsdPrim& prim, MObject pare
-            MObject      tempRefNode = refIter.item();
-            MFnReference tempRefFn(tempRefNode);
-            if (!tempRefFn.isFromReferencedFile()) {
-
-                // Get the name of the reference node so we can check if this node
-                // matches the prim we are processing.  Strip off the "_RN" suffix.
-                //
-                MString refNodeName = tempRefFn.name();
-                MString refName(refNodeName.asChar(), refNodeName.numChars() - 3);
-
-                // What reference node name is the prim expecting?
-                MString expectedRefName = refNameFromPath(parentDag);
-
-                // If found a match, reconnect the reference node's `associatedNode`
-                // attr before loading it, since the previous connection may be gone.
-                //
-                if (refName == expectedRefName) {
-                    // Reconnect the reference node's `associatedNode` attr before
-                    // loading it, since the previous connection may be gone.
-                    connectReferenceAssociatedNode(parentDag, tempRefFn);
-                    refNode = tempRefNode;
-                    break;
+        if (isMayaReference(prim)) {
+            // New behaviour for MayaReference
+            if (prim.HasAttribute(MayaReferenceAttribute)) {
+                UsdAttribute attr = prim.GetAttribute(MayaReferenceAttribute);
+                VtValue      value;
+                attr.Get(&value);
+                auto resInfo = attr.GetResolveInfo();
+                // Check if attribute was directly on prim or inherited.
+                bool isAttributeOnPrim = resInfo.GetNode().GetPath() == prim.GetPath();
+                if (isAttributeOnPrim && value.GetType() == SdfValueTypeNames->String.GetType()) {
+                    MString expectedValue = MString(value.Get<std::string>().c_str());
+                    for (MItDependencyNodes refIter(MFn::kReference); !refIter.isDone();
+                         refIter.next()) {
+                        MObject      tempRefNode = refIter.item();
+                        MFnReference tempRefFn(tempRefNode);
+                        if (!tempRefFn.isFromReferencedFile()) {
+                            if (expectedValue == tempRefFn.name()) {
+                                // Reconnect the reference node's `associatedNode` attr before
+                                // loading it, since the previous connection may be gone.
+                                connectReferenceAssociatedNode(parentDag, tempRefFn);
+                                refNode = tempRefNode;
+
+                                // Update Maya Ref Node on reconnect
+                                MString uniqueRefNodeName
+                                    = getUniqueRefNodeName(prim, parentDag, tempRefFn);
+                                tempRefFn.setName(uniqueRefNodeName);
```

Comment:
I believe that this will fail if the node is locked which it is sometimes.

Reply:
In the case I tested, the rename was working. Need to validate with more test cases.